Source/Particles/NamedComponentParticleContainer.H

```diff
@@ -0,0 +1,127 @@
+#ifndef NamedComponentParticleContainer_H_
```

Comment:
@ax3l 
Here, should we split this file into a `_fwd.H` file and a regular `.H` file?

Reply:
yes, if you have a use case where you can already use `NamedComponentParticleContainer<T>` without accessing its members?

Being able to use `PIdx` on its own is already worth it, I think.